src/it/scala/com/wavesplatform/it/TransferSending.scala

```diff
@@ -0,0 +1,53 @@
+package com.wavesplatform.it
+
+import com.wavesplatform.it.TransferSending.Req
+import com.wavesplatform.it.api.NodeApi.Transaction
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.Future
+import scala.util.Random
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object TransferSending {
+  case class Req(source: String, targetAddress: String, amount: Long, fee: Long)
+}
+
+trait TransferSending {
+  def nodes: Seq[Node]
+
+  def generateRequests(n: Int, balances: mutable.Map[String, Long]): Seq[Req] = {
+    val addresses = nodes.map(_.address)
+    val sourceAndDest = (1 to n).map { _ =>
+      val Seq(src, dest) = Random.shuffle(addresses).take(2)
+      (src, dest)
+    }
+    val (_, requests) = sourceAndDest.foldLeft((balances, ArrayBuffer.empty[Req])) {
+      case ((b, reqs), (src, dest)) =>
```

Comment:
what is the point of making fold if the collection is mutable? 
Either make it immutable or just do `map` here if you wish to keep this mutable.